### DISCUSSION TITLE
convert all gifs in folder

### DISCUSSION THREAD

**Discussion by pavelu85** [ImageMagick/ImageMagick#2218] [Help] [answered]
Hi, I'd like to do same operation for all (gif) files in the folder, for single file I use: convert filename -loop 0 -dispose previous
but instead of filename I'd like to use something like *gif but that gives and error. Is this possible with IM7?

reading the help I see that I must perhaps specify all files separately: convert.exe [options ...] file [ [options ...] file ...]
I tried but this didn't work:
convert -loop 0 -dispose previous test1.gif test11.gif -loop 0 -dispose previous test2.gif test21.gif
where test1.gif and test2.gif are input files

**Comment by snibgo** [comment, marked as answer]
"convert" will read all the input images, and do whatever you want with those.

But you want to process each GIF separately. "mogrify" will do that.

Alternatively, use a shell "for" that loops through the GIFs and calls "convert" once for each GIF.

**Reply by pavelu85** [reply]
mogrify did nicely and the shell loop should work as well indeed. Thanks a lot my friend!
